libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp

```diff
@@ -699,43 +740,42 @@ bool AP_InertialSensor_MPU6000::_check_raw_temp(int16_t t2)
     return (abs(t2 - _raw_temp) < 400);
 }
 
-bool AP_InertialSensor_MPU6000::_block_read(uint8_t reg, uint8_t *buf,
+bool AP_InertialSensor_Invensense::_block_read(uint8_t reg, uint8_t *buf,
                                             uint32_t size)
 {
     return _dev->read_registers(reg, buf, size);
 }
 
-uint8_t AP_InertialSensor_MPU6000::_register_read(uint8_t reg)
+uint8_t AP_InertialSensor_Invensense::_register_read(uint8_t reg)
 {
     uint8_t val = 0;
     _dev->read_registers(reg, &val, 1);
     return val;
 }
 
-void AP_InertialSensor_MPU6000::_register_write(uint8_t reg, uint8_t val, bool checked)
+void AP_InertialSensor_Invensense::_register_write(uint8_t reg, uint8_t val, bool checked)
 {
     _dev->write_register(reg, val, checked);
 }
 
 /*
   set the DLPF filter frequency. Assumes caller has taken semaphore
  */
-void AP_InertialSensor_MPU6000::_set_filter_register(void)
+void AP_InertialSensor_Invensense::_set_filter_register(void)
 {
     uint8_t config;
 
-#if MPU6000_EXT_SYNC_ENABLE
+#if Invensense_EXT_SYNC_ENABLE
     // add in EXT_SYNC bit if enabled
     config = (MPUREG_CONFIG_EXT_SYNC_AZ << MPUREG_CONFIG_EXT_SYNC_SHIFT);
 #else
     config = 0;
 #endif
 
-
     if (enable_fast_sampling(_accel_instance)) {
-        _fast_sampling = (_is_icm_device && _dev->bus_type() == AP_HAL::Device::BUS_TYPE_SPI);
+        _fast_sampling = (_mpu_type > Invensense_MPU6000 && _dev->bus_type() == AP_HAL::Device::BUS_TYPE_SPI);
```

Comment:
It's not great to use a greater than comparison for enum values. Up to you, but this would look a tiny bit better with a `!=`.